### ISSUE TITLE
Trying to get property of non-object   

### ISSUE THREAD

**Issue by ima-tech** [dustin10/VichUploaderBundle#385]
Here's a really strange situation. If I run this line of code on my mac, I have no problem whatsoever. However, if I run the same command line on my server, I get the following error:

I am really clueless on what in this class can cause this error. I've shown you my vich uploader configuration in here:

```
db_driver: orm # or mongodb or propel or phpcr
storage: gaufrette
mappings:
    thumbnail:
        uri_prefix: https://s3.amazonaws.com/%amazon_s3_videos_bucket_name% # you'll need this set to use the Vich URL generator
        upload_destination: thumbnail_storage_fs #Represents the filesystem
        namer: vich_uploader.namer_uniqid
        delete_on_remove: true
        delete_on_update: true
        inject_on_load:   false

    video_file:
        uri_prefix: https://s3.amazonaws.com/%amazon_s3_videos_bucket_name% # you'll need this set to use the Vich URL generator
        upload_destination: video_file_storage_fs #Represents the filesystem
        namer: vich_uploader.namer_uniqid
        delete_on_remove: true
        delete_on_update: true
        inject_on_load:   false
```

My config on my mac (working)
Mac OS X Yosemite
PHP 5.4.34

My config on my server (not working):
Ubuntu 12.04
Apache: 2.4.12
PHP: 5.5.21

Here is the error:

```
/tmp/imatech/osc_web# php app/console cache:clear -v



  [Symfony\Component\Debug\Exception\ContextErrorException]  
  Notice: Trying to get property of non-object               



Exception trace:
 () at /tmp/imatech/osc_web/vendor/vich/uploader-bundle/Metadata/MetadataReader.php:80
 Symfony\Component\Debug\ErrorHandler->handleError() at     /tmp/imatech/osc_web/vendor/vich/uploader-bundle/Metadata/MetadataReader.php:80
 Vich\UploaderBundle\Metadata\MetadataReader->getUploadableFields() at /tmp/imatech/osc_web/vendor/vich/uploader-bundle/DependencyInjection/Compiler/RegisterPropelModelsPass.php:32
 Vich\UploaderBundle\DependencyInjection\Compiler\RegisterPropelModelsPass->process() at /tmp/imatech/osc_web/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/Compiler.php:117
 Symfony\Component\DependencyInjection\Compiler\Compiler->compile() at /tmp/imatech/osc_web/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:614
 Symfony\Component\DependencyInjection\ContainerBuilder->compile() at /tmp/imatech/osc_web/app/bootstrap.php.cache:2565
 Symfony\Component\HttpKernel\Kernel->initializeContainer() at /tmp/imatech/osc_web/app/bootstrap.php.cache:2344
 Symfony\Component\HttpKernel\Kernel->boot() at /tmp/imatech/osc_web/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:70
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /tmp/imatech/osc_web/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at /tmp/imatech/osc_web/app/console:27 
```

Basically, this is causing me problems since I can't run any Symfony lines

Edit1: 
If I remove the bundle form AppKernel, the command can execute without any problems.

**Comment by K-Phoen** [comment]
Looks a lot like #381 

Could you paste your entity and your configuration mapping ? (in a gist, it would be perfect. Or at least using markdown code blocks)

Thanks !

**Comment by ima-tech** [comment]
Here's the first one: https://gist.github.com/ima-tech/4afe3ecbb6a186a71820
Here's the second mapping: https://gist.github.com/ima-tech/b6e790145b7a77eafc74

**Comment by ima-tech** [comment]
@K-Phoen Do you have an idea on what could cause that? And why it could work on one platform but not the other ? Also, it says that there is a problem with RegisterPropelModelsPass, but I'm using ORM and not propel...

**Comment by K-Phoen** [comment]
Nop, no idea yet. I'll try to look into it this weekend :)

**Comment by K-Phoen** [comment]
Can't reproduce the bug, could you try to reproduce it in [the sandbox](https://github.com/K-Phoen/Vich-Uploader-Sandbox/tree/doctrine-sonata)

**Comment by yaK2manD** [comment]
I had the same problem.
I use MongoDB and my mapping was in yml file (DirectoryInsideDocumentFolder.MyDocument.yml). My document were not directly inside Document folder.

I came back to annotation '@Vich' and it works. As @ima-tech said, it uses RegisterPropelModelsPass. Maybe it comes from there.

**Comment by kevinpostma** [comment]
I had the same problem to recently, In my case it was because i had edited the filename_property in my entity.xml in the vich_uploader xml config. but the cache was not updated. and couldnt be updated because all the console commands broke because a invalid cache.

After i cleared the cache manually by executing rm -rf app/cache/* everything worked fine again.

**Comment by gonzalovilaseca** [comment]
I think the problem occurs when the entity is placed outside the bundle.

In line 33 of `Vich\UploaderBundle\Metadata\Driver\FileLocator`:
``
if ('' !== $prefix && 0 !== strpos($class->getNamespaceName(), $prefix)) {
``
So unless the entity is in the same namespace as vich mapping definition file, `strpos` fails.

I'll do a PR shortly to fix this.

**Comment by GregoireHebert** [comment]
I got the same problem because the config file placed in `MyBundle/Resources/config/vich_uploader/MyClassName.yml` was named after an entity that has been renamed...

So `MyBundle/Entity/MyClassName.php` became `MyBundle/Entity/MyNewClassName.php`
and i forgot to rename `MyBundle/Resources/config/vich_uploader/MyClassName.yml` into
`MyBundle/Resources/config/vich_uploader/MyNewClassName.yml`

Check if the two are spelled the same and dont forget the clear the cache.

**Comment by garak** [comment]
We likely need to do some check in [`getUploadableFields()`](https://github.com/dustin10/VichUploaderBundle/blob/master/Metadata/MetadataReader.php#L77) to see if `$metadata` is null

**Comment by dorcyv** [comment]
Any news about that ?

**Comment by paul-mesnilgrente** [comment]
I am having the same issue when I run `php bin/console doctrine:generate:entities AppBundle:NailArtPhoto`:

```text
In ClassMetadataInfo.php line 3325:
                                                       
  Notice: Trying to get property 'name' of non-object
```

I tried to clean my cache with `rm -rf var/cache` but it doesn't change anything.

Here is my `app/config/config.yml`:
```yaml
vich_uploader:
    db_driver: orm
    mappings:
        nailart_photo:
            uri_prefix: /nail-arts/photos
            upload_destination: '%kernel.project_dir%/uploads/nailarts/'
```

And my class:

```php
<?php

namespace AppBundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\HttpFoundation\File\File;
use Symfony\Component\HttpFoundation\File\UploadedFile;
use Vich\UploaderBundle\Entity\File as EmbeddedFile;
use Vich\UploaderBundle\Mapping\Annotation as Vich;

/**
 * @ORM\Entity
 * @Vich\Uploadable
 */
class NailArtPhoto
{
    /**
     * @ORM\Id
     * @ORM\Column(type="integer")
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * NOTE: This is not a mapped field of entity metadata, just a simple property.
     * 
     * @Vich\UploadableField(mapping="nailart_photo", fileNameProperty="image.name", size="image.size", mimeType="image.mimeType", originalName="image.originalName", dimensions="image.dimensions")
     * 
     * @var File
     */
    private $imageFile;

    /**
     * @ORM\Embedded(class="Vich\UploaderBundle\Entity\File")
     *
     * @var EmbeddedFile
     */
    private $image;

    /**
     * @ORM\Column(type="datetime")
     *
     * @var \DateTime
     */
    private $updatedAt;
    
    public function __construct()
    {
        $this->image = new EmbeddedFile();
    }

    /**
     * If manually uploading a file (i.e. not using Symfony Form) ensure an instance
     * of 'UploadedFile' is injected into this setter to trigger the  update. If this
     * bundle's configuration parameter 'inject_on_load' is set to 'true' this setter
     * must be able to accept an instance of 'File' as the bundle will inject one here
     * during Doctrine hydration.
     *
     * @param File|UploadedFile $image
     */
    public function setImageFile(?File $image = null)
    {
        $this->imageFile = $image;

        if (null !== $image) {
            // It is required that at least one field changes if you are using doctrine
            // otherwise the event listeners won't be called and the file is lost
            $this->updatedAt = new \DateTimeImmutable();
        }
    }

    public function getImageFile(): ?File
    {
        return $this->imageFile;
    }

    public function setImage(EmbeddedFile $image)
    {
        $this->image = $image;
    }

    public function getImage(): ?EmbeddedFile
    {
        return $this->image;
    }
}
```
